cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -347,28 +348,28 @@ public ApplicationWithPrograms deployApp(Id.Namespace namespace, @Nullable Strin
    * @param namespaceId the {@link Id.Namespace} under which all application should be deleted
    * @throws Exception
    */
-  public void removeAll(final Id.Namespace namespaceId) throws Exception {
+  public Iterable<Id.Program> removeAll(final Id.Namespace namespaceId) throws Exception {
```

Comment:
Why return `Iterable`? You can return a `Set`. Also, shouldn't use the deprecated `Id.Program` class. Please use `ProgramId` instead.